components/UI/InviteUserToMatrixRoom.js

```diff
@@ -0,0 +1,192 @@
+    const [isInviteDialogueOpen, setIsInviteDialogueOpen] = useState(false);
+    const [searchInput, setSearchInput] = useState('');
+    const [searchResults, setSearchResults] = useState([]);
+    const [isInviting, setIsInviting] = useState(false);
+    const [isFetchingSearchResults, setIsFetchingSearchResults] = useState(false);
+    const [userFeedback, setUserFeedback] = useState('');
+    const [validUserObject, setValidUserObject] = useState(false);
+
+    const customModalStyles = {
+        content: {
+            top: '50%',
+            right: 'auto',
+            bottom: 'auto',
+            left: '50%',
+            minWidth: '60%',
+            padding: 'calc(var(--margin) * 2)',
+            marginRight: '-50%',
+            transform: 'translate(-50%, -50%)',
+        },
+    };
+
+    const { t } = useTranslation();
+
+    const handleClick = () => {
+        setIsInviteDialogueOpen(prevState => !prevState);
+    };
+
+    const handleChange = (event) => {
+        setSearchInput(event.target.value);
+        debouncedFetchUsersForContributorSearch(event.target.value);
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const debouncedFetchUsersForContributorSearch = useCallback(debounce((val) => fetchUsersForContributorSearch(val), 300), []);
+
+    const fetchUsersForContributorSearch = useCallback(async (a) => {
+        setIsFetchingSearchResults(true);
+        try {
+            const users = await matrixClient.searchUserDirectory({ term: a });
+            // we only update the state if the returned array has entries, to be able to check if users a matrix users or not further down in the code (otherwise the array gets set to [] as soon as you selected an option from the datalist)
+            users.results.length > 0 && setSearchResults(users.results);
+        } catch (err) {
+            console.error('Error whhile trying to fetch users: ' + err);
+        } finally {
+            setIsFetchingSearchResults(false);
+        }
+    }, [matrixClient]);
+
+    const handleInvite = async (e) => {
+        setIsInviting(true);
+        e?.preventDefault();
+
+        function clearInputs() {
+            setUserFeedback('');
+            setSearchInput('');
+            setIsInviting(false);
+        }
+
+        await matrixClient.invite(roomId, validUserObject.userId)
+            .catch(async err => {
+                // if something went wrong we display the error and clear all inputs
+                setUserFeedback(<ErrorMessage>{ err.data?.error }</ErrorMessage>);
+                await new Promise(() => setTimeout(() => {
+                    clearInputs();
+                }, 3000));
+
+                return;
+            });
+
+        // if everything is okay, we let the user know and exit the modal view.
+        setUserFeedback('✓ ' + validUserObject.displayName + t(' was invited and needs to accept your invitation'));
```

Comment:
`validUserObject.displayName` is empty when I invite someone by manually putting in their Matrix ID without using the search-autocomplete:
<img width="602" alt="image" src="https://github.com/medienhaus/medienhaus-spaces/assets/706419/842bbc68-5777-4c53-868a-11e59d2bf0d9">